tests/ui/from_instead_of_into.rs

```diff
@@ -0,0 +1,29 @@
+// run-rustfix
+#![allow(unused_imports)]
+#![warn(clippy::from_instead_of_into)]
+use std::convert::TryFrom;
+
+fn foo<T>(a: T)
+where
+    u32: From<T>,
+{
+}
+
+fn foo1<T>(a: T)
+where u32: Copy + Clone + From<T>,
```

Comment:
Another predicate here. Is it not enough here?

I can't imagine a more complicated predicate that is also meaningful here. My understanding of generic bounds is rather shallow, I didn't know how to deepen my understanding, I have tried to search `rust generic bound` but didn't get progress here. How do you suggest I can learn about it? How did you learn about generic bounds in the past?